controllers/snapshot/snapshot_adapter.go

```diff
@@ -129,7 +129,8 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 // EnsureGlobalComponentImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
 // being updated when the ApplicationSnapshot passed all the integration tests
 func (a *Adapter) EnsureGlobalComponentImageUpdated() (results.OperationResult, error) {
-	if (a.component != nil) && gitops.HaveHACBSTestsSucceeded(a.snapshot) {
+	if (a.component != nil) && gitops.HaveHACBSTestsSucceeded(a.snapshot) &&
+		a.snapshot.Labels[gitops.PipelineAsCodeEventTypeLabel] == gitops.PipelineAsCodePushType {
```

Comment:
I think we'll need to also check if the snapshot has the `gitops.PipelineAsCodeEventTypeLabel` label at all. Might be best to create a new function that checks if the label is the `push` type and returns a boolean if it is.

Reply:
OK. Updated, thanks!

Reply:
Thanks, nicely implemented!